pkg/k8s/watchers/pod.go

```diff
@@ -475,6 +485,15 @@ func (k *K8sWatcher) deleteK8sPodV1(pod *slim_corev1.Pod) error {
 		k.redirectPolicyManager.OnDeletePod(pod)
 	}
 
+	podNSName := k8sUtils.GetObjNamespaceName(&pod.ObjectMeta)
+	podEP := k.endpointManager.LookupPodName(podNSName)
```

Comment:
I think this can be susceptible to race conditions? `LookupPodName` acquires a lock before returning a pointer to the endpoint struct. But thereafter you don't acquire the lock when you read the endpoint's fields below?

Reply:
I'm a bit confused how this would incur a race condition? I mean `LookupPodName` enters critical section and exits upon return, so using the pointer afterwards should be safe?

Reply:
I see, so my initial thought is that the modification of `podEP.NoTrackPort` is handled only when the pod annotation changes, and I was under the assumption that if the pod is being deleted, no other thread should be altering its annotation, but I see that it could happen in theory.

I agree removing the check is a good approach, that was placed there for some optimization (so we don't have to enqueue an update event if no NOTRACK rules are present).